crates/vm/levm/src/opcode_handlers/system.rs

```diff
@@ -43,11 +43,20 @@ impl VM {
             return Err(VMError::OpcodeNotAllowedInStaticContext);
         }
 
-        let memory_byte_size = (args_offset + args_size).max(ret_offset + ret_size);
+        let memory_byte_size = (args_offset
+            .checked_add(args_size)
+            .ok_or(VMError::VeryLargeNumber)?)
+        .max(
+            ret_offset
+                .checked_add(ret_size)
+                .ok_or(VMError::VeryLargeNumber)?,
+        );
```

Comment:
Nit. The parenthesis is not needed (do not commit this, it needs to be formatted).
```suggestion
        let memory_byte_size = args_offset
            .checked_add(args_size)
            .ok_or(VMError::VeryLargeNumber)?
        .max(
            ret_offset
                .checked_add(ret_size)
                .ok_or(VMError::VeryLargeNumber)?,
        );
```